dokka-runners/dokka-gradle-plugin/src/main/kotlin/DokkaExtension.kt

```diff
@@ -213,5 +214,86 @@ constructor(
      */
     @Deprecated("Moved to DokkaPublication#suppressObviousFunctions")
     abstract val suppressObviousFunctions: Property<Boolean>
+
+    /**
+     * JSON configuration of Dokka plugins is deprecated.
+     * Typesafe configuration must be used instead - see [pluginsConfiguration].
+     *
+     * In DPGv1 the Dokka plugins could be configured by manually writing JSON.
+     * This caused issues with registering task inputs for Gradle up-to-date checks.
+     * (For more information on registering task inputs, see
+     * [Gradle Docs: Incremental build](https://docs.gradle.org/current/userguide/incremental_build.html)).
+     *
+     * In DGPv2 Dokka plugins must be configured in a typesafe way, using [pluginsConfiguration].
+     *
+     * #### Example
+     *
+     * This example demonstrates how to convert JSON configuration of the
+     * [Dokka Versioning plugin](https://kotl.in/dokka-versioning-plugin)
+     * into the new, typesafe config.
+     *
+     * ```
+     * // Deprecated configuration of the Dokka Versioning plugin:
+     * tasks.dokkaHtmlMultiModule {
+     *     pluginsMapConfiguration.set(
+     *         mapOf(
+     *             "org.jetbrains.dokka.versioning.VersioningPlugin" to """
+     *                 { "version": "1.2", "olderVersionsDir": "$projectDir/dokka-docs" }
+     *             """.trimIndent()
+     *         )
+     *     )
+     * }
+     *
+     * // New configuration in DGPv2 is typesafe and compatible with incremental builds.
+     * dokka {
+     *     pluginsConfiguration {
+     *         versioning {
+     *             version.set("1.2")
+     *             olderVersionsDir.set(projectDir.resolve("dokka-docs"))
+     *         }
+     *     }
+     * }
+     * ```
+     *
+     * #### Dokka Plugin authors
```

Comment:
```suggestion
     * #### External Dokka Plugin configuration
```

Probably it would be nice to put all of this into migration guide, and then just add a link to it. Or copy-paste. I don't have preferences, but I think it should be in migration guide.

Also, is needed not only for plugin authors, but initially mostly for plugin consumers.

Reply:
Let's merge this PR as-is and we can update the docs separately?

https://youtrack.jetbrains.com/issue/KT-72447/Document-Dokka-plugin-configuration

I think some of this docs can go into the Dokka dev docs too.

Reply:
ok
dev docs could be updated later (after release)